### DISCUSSION TITLE
How can I point my search to only search with in a certain document library?

### DISCUSSION THREAD

**Discussion by KhaledXElsayed** [microsoft-search/pnp-modern-search#2797] [General]
Hello all,

How can I point my search to only search within a specific document library? I tried it using the path command but was only able to use path={site]

How can I add the URL for a document library? Please help

**Comment by kasperbolarsen** [comment]
path:"https://siteURL.com/DocumentLibraryName/*"

This is an excellent book on the subject,  https://www.amazon.com/SharePoint-Search-Queries-Explained-writing-ebook/dp/B00RO84UBQ/ref=sr_1_1?qid=1677090725&refinements=p_27%3AMikael+Svenson&s=books&sr=1-1

**Reply by KhaledXElsayed** [reply]
Found the answer, thank you!

create a search query like {searchboxquery}path:"https://contoso.sharepoint.com/sites/sitename/libraryname" (contentclass:STS_ListItem OR IsDocument:True) and then use [the PnP Modern Search Web Parts ](https://techcommunity.microsoft.com/t5/microsoft-365-pnp-blog/using-the-pnp-modern-search-web-parts/ba-p/2239771) on the modern SharePoint page to customize the search experience.

**Comment by KhaledXElsayed** [comment]
Found the answer!
create a search query like {searchboxquery}path:"https://contoso.sharepoint.com/sites/sitename/libraryname" (contentclass:STS_ListItem OR IsDocument:True) and then use [the PnP Modern Search Web Parts ](https://techcommunity.microsoft.com/t5/microsoft-365-pnp-blog/using-the-pnp-modern-search-web-parts/ba-p/2239771) on the modern SharePoint page to customize the search experience.

**Reply by influential-eliot** [reply]
How would one apply that to multiple libraries, @KhaledXElsayed ?

Would I use something like this?
```
{searchboxquery} (path:"https://contoso.sharepoint.com/sites/sitename/libraryname1" OR path:"https://contoso.sharepoint.com/sites/sitename/libraryname2" OR path:"https://contoso.sharepoint.com/sites/sitename/libraryname3" ) (contentclass:STS_ListItem OR IsDocument:True)
```

**Reply by kasperbolarsen** [reply]
That looks just right :-) If you want to try out various queries I can recommend using one of the tools here https://microsoft-search.github.io/pnp-modern-search/using-query-tools-to-verify-issues/

**Reply by influential-eliot** [reply]
Cheers, @kasperbolarsen !

I tried it and got it right, aye ... but then I tried to limit to just folders using things like this:
```
-ContentTypeId:0x0120* IsDocument:1
```
... and / or this:
```
-ContentType:folder AND -contentclass: STS_List_DocumentLibrary
```
... and that basically didn't work. 😅 

I couldn't find just a simple list of terms / operators (etc) to use to build my query template with, which was a bit frustrating.

**Reply by kasperbolarsen** [reply]
Are you looking for just the folders or the content in the folders?

**Reply by influential-eliot** [reply]
I'm looking to exclude the folders (and any libraries I suppose) as results in and of themselves.

If there were a list of usable terms, and operators ... or even a query builder, rather than a query tester ... I'd not be bothering you ... sorry about this. (i have seen [this](https://learn.microsoft.com/en-us/sharepoint/dev/general-development/keyword-query-language-kql-syntax-reference), yes)

EDIT - This ... might(?) have worked:
```
{searchboxquery}
(
 ( -ContentType:folder AND IsDocument:1 AND path:"https://contoso.sharepoint.com/sites/MuraGroup/library1" ) 
 OR 
 ( -ContentType:folder AND IsDocument:1 AND path:"https://contoso.sharepoint.com/sites/MuraGroup/library2" ) 
 OR 
 ( -ContentType:folder AND IsDocument:1 AND path:"https://contoso.sharepoint.com/sites/MuraGroup/library3") 
) 
AND 
(
 contentclass:STS_ListItem 
 OR 
 IsDocument:True 
 OR 
 -ContentType:folder
) 
NOT 
 ContentType:folder 
```
I appreciate that there is some redundancy in there ... but this wasn't working ... then I saved a draft, went back, and suddenly it now does ... I _think_.

**Reply by kasperbolarsen** [reply]
search isn't that simple, and some of us has made it into a living ;-) This book is an exellent place to start https://www.techmikael.com/2015/01/sharepoint-search-queries-explainedthe.html
